### PR TITLE
ARROW-2445: [Rust] Add documentation and make some fields private

### PR DESCRIPTION
**Pull request by andygrove**
No functional changes.

### CODE REVIEW

File: rust/src/array.rs [hunk 1]
```diff
@@ -69,10 +69,14 @@ arraydata_from_primitive!(u32, UInt32);
...
-    pub validity_bitmap: Option<Bitmap>,
-    pub data: ArrayData,
+    /// number of elements in the array
+    len: i32,
+    /// number of null elements in the array
+    null_count: i32,
+    /// If null_count is greater than zero then the validity_bitmap will be Some(Bitmap)
+    validity_bitmap: Option<Bitmap>,
+    /// The array of elements
+    data: ArrayData,
```
Comment (liurenjie1024) [inline rust/src/array.rs]: Add a field for data type?

File: rust/src/array.rs [hunk 2]
```diff
@@ -86,13 +90,25 @@ impl Array {
...
         self.len as usize
     }
+
+    /// number of null elements in the array
+    pub fn null_count(&self) -> usize {
+        self.null_count as usize
+    }
+
+    /// If null_count is greater than zero then the validity_bitmap will be Some(Bitmap)
+    pub fn validity_bitmap(&self) -> &Option<Bitmap> {
```
Comment (liurenjie1024) [inline rust/src/array.rs]: I don't think it's a good idea to expose validity bitmap directly to user. I think we should just add methods to test whether a position is set.

Comment (xhochy) [reply rust/src/array.rs]: We have seen some cases in C++ that it is beneficial in a lot of cases to work directly on the bitmap as the compiler can drastically speedup access to it using SIMD instructions.

**Review by xhochy — APPROVED** [review]

File: rust/src/array.rs (hunk 1 above)
Comment (andygrove) [reply rust/src/array.rs]: Array is the physical memory storage and is separate from the meta-data describing the types.

**Comment by andygrove** [comment]
@pitrou or @xhochy Could you please merge these changes?

**Pull request closed without merging**